lib/improver/tests/standardise/test_StandardiseGridAndMetadata.py

```diff
@@ -0,0 +1,191 @@
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for the standardise.StandardiseGridAndMetadata plugin."""
+
+import unittest
+import numpy as np
+from iris.tests import IrisTest
+
+from improver.standardise import StandardiseGridAndMetadata
+from improver.tests.set_up_test_cubes import set_up_variable_cube
+
+
+class Test__init__(unittest.TestCase):
+    """Test initialisation"""
+
+    def test_default(self):
+        """Test initialisation with default options"""
+        plugin = StandardiseGridAndMetadata()
+        self.assertEqual(plugin.regrid_mode, 'bilinear')
+        self.assertEqual(plugin.extrapolation_mode, 'nanmask')
+        self.assertIsNone(plugin.landmask_source_grid)
+        self.assertIsNone(plugin.landmask_vicinity)
+        self.assertEqual(plugin.landmask_name, 'land_binary_mask')
+        self.assertSequenceEqual(
+            plugin.grid_attributes,
+            ['mosg__grid_version', 'mosg__grid_domain', 'mosg__grid_type',
+             'mosg__model_configuration', 'institution'])
+
+    def test_error_missing_landmask(self):
+        """Test an error is thrown if no mask is provided for masked
+        regridding"""
+        msg = "requires an input landmask cube"
+        with self.assertRaisesRegex(ValueError, msg):
+            StandardiseGridAndMetadata(regrid_mode='nearest-with-mask')
+
+
+class Test_process(IrisTest):
+    """Test the process method. Complex regridded values are not tested here
+    as this is covered by unit tests for the regridding routines
+    (iris.cube.Cube.regrid and improver.utilities.spatial.RegridLandAndSea).
+    """
+
+    def setUp(self):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method name "setUp" doesn't conform to '((test_[a-zA-Z0-9_]+)|([a-z_][a-z0-9_]{2,30}))$' pattern](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4508565)